# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/sauerbraten/hotfix-new.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

Use the standard placement new, <cmath> includes <new> with GCC 16.

--- sauerbraten.orig/src/shared/tools.h	2020-11-30 02:47:28.000000000 +0100
+++ sauerbraten/src/shared/tools.h	2026-09-27 19:17:58.484311652 +0200
@@ -36,10 +36,7 @@ typedef unsigned long long int ullong;
 
 void *operator new(size_t, bool);
 void *operator new[](size_t, bool);
-inline void *operator new(size_t, void *p) { return p; }
-inline void *operator new[](size_t, void *p) { return p; }
-inline void operator delete(void *, void *) {}
-inline void operator delete[](void *, void *) {}
+#include <new>
 
 #ifdef swap
 #undef swap
